extends CanvasLayer

#questions
var trivia = []
var trivia_JP = []
var trivia_labels = []
var current_trivia_selection = 0

#focus management
var buttons = []
var current_button_index = 0

var disable_buttons = true


#pick randomly an answer 
func _on_answer5_pressed():
	if trivia[current_trivia_selection]["a"].size() == 0:
		return null
	randomize()
	var random_index = randi() % trivia[current_trivia_selection]["a"].size()
	_on_button_pressed(random_index)
	
func _ready() -> void:
	
	
	#random color blocks
	
	
	#manage focus with joypad
	buttons = [%Answer1, %Answer2, %Answer3, %Answer4]
		
	#Q1
	trivia.append({
	"q" : "âœ¦ WHAT'S YOUR FAVORITE FRUIT?",
	"a" : ["ğŸŠ\nORANGE", "ğŸ“\nSTRAWBERRY", "ğŸŒ\nBANANA", "ğŸ‰\nWATERMELON"]
	})
	#Q2
	trivia.append({
	"q" : "âœ¦ ARE YOU MORE?",
	"a" : ["ADVENTUROUS", "COZY BEHIND A DESK", "DREAMING ABOUT SCIFI", "NOT REALLY INTO ANYTHING"]
	})
	#Q3
	trivia.append({
	"q" : "âœ¦ HOW MUCH OF A GAMER ARE YOU?",
	"a" : ["BRING IT ON, I LOVE SUPER HARD GAMES!", "I'M MODERATELY INTO GAMES", "I ENJOY MORE WATCHING", "NOT A PLAYER, I'M HERE FOR THE INSIGHTS"]
	})
	
	## -- JP support
	trivia_JP.append({
	"q" : "âœ¦ å¥½ããªæœç‰©ã¯ä½•ã§ã™ã‹ï¼Ÿ",
	"a" : ["ğŸŠ\nã‚ªãƒ¬ãƒ³ã‚¸", "ğŸ“\nã‚¤ãƒã‚´", "ğŸŒ\nãƒãƒŠãƒŠ", "ğŸ‰\nã‚¹ã‚¤ã‚«"]
	})
	#Q2
	trivia_JP.append({
	"q" : "âœ¦ ã©ã¡ã‚‰ã‹ã¨è¨€ãˆã°",
	"a" : ["ã‚¢ã‚¯ãƒ†ã‚£ãƒ–ãªæ–¹ã§ã™ã‹", "ãŠå®¶ã§ã‚†ã£ãã‚Šéã”ã™ã®ã‹", "SFã®ä¸–ç•Œã‚’å¦„æƒ³ã™ã‚‹ã®ãŒ", "ç‰¹ã«ãƒãƒã£ã¦ã„ã‚‹ã“ã¨ã¯ã‚ã‚Šã¾ã›ã‚“"]
	})
	#Q3
	trivia_JP.append({
	"q" : "âœ¦ ã‚²ãƒ¼ãƒ ã¯ã©ã®ç¨‹åº¦ã‚„ã‚Šã¾ã™ã‹ï¼Ÿ",
	"a" : ["ã‹ã‹ã£ã¦ã“ã„ï¼æ¿€ãƒ ã‚ºã®ã‚²ãƒ¼ãƒ ãŒå¤§å¥½ãã ï¼", "ã‚²ãƒ¼ãƒ ã¯ãã“ãã“ã‚„ã‚Šã¾ã™ã€‚", "è‡ªåˆ†ã§ã‚„ã‚‹ã‚ˆã‚Šã€è¦‹ã‚‹æ–¹ãŒå¥½ãã§ã™ã€‚", "ã‚²ãƒ¼ãƒ ã¯ã—ãªã„ã®ã§ã™ãŒã€ãƒ‡ãƒ¼ã‚¿åˆ†æã®æ–¹ãŒèˆˆå‘³ã‚ã‚Šã¾ã™ã€‚"]
	})
	
	#--default: EN
	
	##--- JP
	if SignalBus.language == "JP":
		trivia_labels = trivia_JP
		%Answer5.text = "é©šã‹ã›ã¦ä¸‹ã•ã„ï¼"
	else:
		trivia_labels = trivia
		%Answer5.text = "JUST SURPRISE ME"
	set_button_texts(trivia_labels[current_trivia_selection])	
			
	buttons[0].grab_focus()
	
	# Connect signals for all buttons
	for i in range(buttons.size()):
		buttons[i].pressed.connect(_on_button_pressed.bind(i))
	
	await get_tree().create_timer(2.0).timeout  # Wait time for cooldown
	disable_buttons = false
	
func _process(_delta):
	#manage focus with joypad
	if Input.is_action_just_pressed("attack"):
		_on_button_pressed(current_button_index)
		
	if Input.is_action_just_pressed("ui_right"):
		# Move to next button
		current_button_index = min(current_button_index + 1, buttons.size() - 1)
		buttons[current_button_index].grab_focus()
	elif Input.is_action_just_pressed("ui_left"):
		# Move to previous button
		current_button_index = max(current_button_index - 1, 0)
		buttons[current_button_index].grab_focus()	

func set_button_texts(qa_array):
	#print( qa_array['q'])
	%Question.text = qa_array['q']
	#buttons[0].grab_focus()
	for i in qa_array['a'].size():
		#print( qa_array['a'][i])
		buttons[i].text = qa_array['a'][i]

func _on_button_pressed(button_index):
	if !disable_buttons:
		disable_buttons = true
		#print("Question " + trivia[current_trivia_selection]['a'][button_index])
		SignalBus.trivia_question_received.emit(trivia[current_trivia_selection]['q'], trivia[current_trivia_selection]['a'][button_index])
		current_trivia_selection += 1
		if current_trivia_selection < trivia.size():
			#print("next question")
			set_button_texts(trivia_labels[current_trivia_selection])
		else: #go back to the game
			#print("finsihed")
			SignalBus.screen_state.emit(SignalBus.CONTROLS)	
		
		await get_tree().create_timer(1.0).timeout  # Wait time for cooldown
		disable_buttons = false
